core/src/main/java/io/substrait/expression/ExpressionCreator.java

```diff
@@ -286,6 +286,19 @@ public static Expression.StructLiteral struct(
     return Expression.StructLiteral.builder().nullable(nullable).addAllFields(values).build();
   }
 
+  public static Expression.StructNested nestedStruct(
+      boolean nullable, Iterable<Expression.Literal> fields) {
```

Comment:
```suggestion
      boolean nullable, Iterable<Expression> fields) {
```
The `fields` param of a `Nested.Struct` takes in an `Expression`, rather than just specifically a `Literal`. 
```proto
    message Struct {
      // Zero or more possibly heterogeneously-typed list of expressions that
      // form the struct fields.
      repeated Expression fields = 1;
    }
```

Is it possible to model the same here?